### PR TITLE
fix: Zapier subscriber url shown in webhooks

### LINKED ISSUE #12555: [CAL-2762] Zapier subscriber url shown in webhooks
When creating a new Zap with the Cal.com trigger, the Zapier subscriber URL is shown in webhooks as if it was a normal webhook. Zapier webhooks should not show up here, they are deleted automatically when the Zap is disabled or deleted. 

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2762](https://linear.app/calcom/issue/CAL-2762/zapier-subscriber-url-shown-in-webhooks)</sub>

### PR DESCRIPTION
**Pull request by riddhesh-mahajan**
## What does this PR do?

To address this issue, the function now utilizes regular expressions to filter out Zapier subscriber URLs from the list of webhooks. The regular expressions are designed to match Zapier subscriber URLs and exclude them from the final list of valid webhooks.

Fixes #12555 

https://www.loom.com/share/2cc8d4e49e5044febe0cf248af58ff24?sid=74c55c47-6394-4580-a2e3-e34f4e32dae3

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?
1) Created a new Zap with the Cal.com trigger.
2) Verified that Zapier subscriber URLs are excluded from the list of webhooks.
3) Checked that valid webhooks are still included in the list.

### CODE REVIEW

File: packages/trpc/server/routers/viewer/webhook/getByViewer.handler.ts [hunk 1]
```diff
@@ -80,7 +89,8 @@ export const getByViewerHandler = async ({ ctx }: GetByViewerOptions) => {
     throw new TRPCError({ code: "INTERNAL_SERVER_ERROR" });
   }
 
-  const userWebhooks = user.webhooks;
+  let userWebhooks = user.webhooks;
+  userWebhooks = userWebhooks.filter(filterWebhooksWithInvalidSubscriptionUrl);
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/webhook/getByViewer.handler.ts:93]: Webhooks have an app field. We can just filter out all webhooks that have zapier in the app field

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
Thank you for taking this issue. I left you a comment, there is an easier way to do that 🙏

File: packages/trpc/server/routers/viewer/webhook/getByViewer.handler.ts (hunk 1 above)
Comment (riddhesh-mahajan) [reply packages/trpc/server/routers/viewer/webhook/getByViewer.handler.ts]: @CarinaWolli  Thank you for your quick review. I've made the suggested changes. Please review the updated code.

File: packages/trpc/server/routers/viewer/webhook/getByViewer.handler.ts [hunk 2]
```diff
@@ -80,7 +89,8 @@ export const getByViewerHandler = async ({ ctx }: GetByViewerOptions) => {
     throw new TRPCError({ code: "INTERNAL_SERVER_ERROR" });
   }
 
-  const userWebhooks = user.webhooks;
+  let userWebhooks = user.webhooks;
+  userWebhooks = userWebhooks.filter(filterWebhooks);
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/webhook/getByViewer.handler.ts:93]: We need to do the same for team webhooks as well

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
Great job in filtering in the user webhooks. Let's do the same for team webhooks as well 🙏

File: packages/trpc/server/routers/viewer/webhook/getByViewer.handler.ts (hunk 2 above)
Comment (riddhesh-mahajan) [reply packages/trpc/server/routers/viewer/webhook/getByViewer.handler.ts:93]: Sure

**Comment by riddhesh-mahajan** [comment]
@CarinaWolli I have added the requested changes. Let me know if any other changes are required.

**Review by CarinaWolli — APPROVED** [review]
Looks good now! Thank you for this fix 🙏

(Failed e2e tests and type checks are not related to you changes)

**Pull request merged**